src/opentelemetry.cr

```diff
@@ -110,13 +115,55 @@ module OpenTelemetry
   # ```
   def self.configure
     yield CONFIG
+
+    if ENV["OTEL_SERVICE_NAME"]?.presence && !CONFIG.service_name.presence
+      CONFIG.service_name = ENV["OTEL_SERVICE_NAME"]
+    end
```

Comment:
> My thought process there is that it'll set it and then if it's set inside the yield it will overwrite that. Does that sound right?

That sounds brilliant. I didn't think of that order to it and love it 👍🏼

Reply:
While I believe the precedence order is not yet stated officially in the spec (for example, there's still [this open issue](https://github.com/open-telemetry/opentelemetry-specification/issues/1171)), I am in Camp Environment Wins Over Code. The quick argument for that is so your have an opportunity to change/update/tweak/correct configuration at runtime without code changes and rebuilds/deploys.

Reply:
@robbkidd That makes sense. It's common to have precedence match specificity (thinking about how in a lot of CLI apps, command-line flags > config file > env), and environment is more specific than the binary.

This would also allow overriding settings for very specific invocations of the binary with the same `config/opentelemetry.cr` (or whatever) file, like DB migrations if the deployment environment allows that.

Reply:
Exactly! 

![A man pointing, "you got it."](https://user-images.githubusercontent.com/517302/225105568-390f29f7-1007-4ea3-abdd-ef7dc435a34a.gif)

Reply:
Opened an issue for this in #4